src/clr-angular/forms/common/if-error/if-error.service.spec.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import {FormControl} from "@angular/forms";
+
+import {NgControlService} from "../providers/ng-control.service";
+
+import {IfErrorService} from "./if-error.service";
+
+
+export default function(): void {
+    describe("IfErrorService", function() {
+        let service, ngControlService, testControl;
+
+        beforeEach(() => {
+            testControl = new FormControl(true);
+            ngControlService = new NgControlService();
+            service = new IfErrorService(ngControlService);
+        });
+
+
+        it("subscribes to the statusChanges when the control is emitted", () => {
+            spyOn(testControl.statusChanges, "subscribe").and.callThrough();
+            ngControlService.control = testControl;
+            expect(testControl.statusChanges.subscribe).toHaveBeenCalled();
+        });
+
+        it("provides observable for statusChanges, passing the control", () => {
+            const cb = jasmine.createSpy("cb");
+            const sub = service.statusChanges.subscribe(control => cb(control));
+            ngControlService.control = testControl;
+            // Change the state of the input to trigger statusChange
+            testControl.markAsTouched();
+            testControl.updateValueAndValidity();
```

Comment:
Would this trigger the `blur` event on the control?

Reply:
I am not sure why we have to call `updateValueAndValidity` because you are just filtering for the `touch` event:

```
this.ngControlService.control.statusChanges.pipe(filter(() => this.ngControlService.control.touched))
                .subscribe(() => {
                  this._statusChanges.next(this.ngControlService.control);
                }));
```

Reply:
I still haven't understood why we need to call `updateValueAndValidity` but I will ask you that question on a call separately.